### PR TITLE
Feature - cypress

### PR DESCRIPTION
**Pull request by joefitter**
* Updated to use cypress for integration tests
* Added integration tests for org mapper

### CODE REVIEW

File: ui/helpers/api.js [hunk 1]
```diff
@@ -31,6 +31,27 @@ export function queriseSelections(selections) {
...
+function getSearchQuery(q) {
+  if (!q) {
+    return undefined;
+  }
+
+  let query = `title:${q}~ OR ${q}`;
+
+  const organisationMappings = {
+    prsb: 'professional-record-standards-body',
+    nhs: 'nhs-digital'
```
Comment (roc) [inline ui/helpers/api.js:43]: Yes! And... I think add a few more here, like nhsx nhsd, "nhs digital"

File: ui/cypress.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "baseUrl": "http://localhost:3000"
```
Comment (lennym) [inline ui/cypress.json:2]: Does Cypress support this being overridden at run time with a command line flag?

File: ui/cypress/fixtures/example.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
Comment (lennym) [inline ui/cypress/fixtures/example.json:5]: If we're not actively using this file, can it be deleted please?

**Comment by lennym** [comment]
There's three random mp4 files, which I think are assets from the test run. If these are not required for the execution of tests can they be a) deleted, and b) the output directory added to `.gitignore`?

File: ui/package.json [hunk 4]
```diff
@@ -7,18 +7,20 @@
...
+    "classnames": "^2.3.1",
     "date-fns": "^2.25.0",
     "lodash": "^4.17.21",
     "marked": "^4.0.5",
     "mdast-util-from-markdown": "^1.2.0",
     "mustache": "^4.2.0",
-    "next": "^11.1.3",
-    "nextjs-breadcrumbs": "^1.1.9",
+    "next": "^12.1.5",
+    "nextjs-breadcrumbs": "^0.9.4",
```
Comment (lennym) [inline ui/package.json:23]: Any reason this has been _downgraded_?

Comment (joefitter) [reply ui/package.json:23]: npm audit fix 🤷

Comment (joefitter) [reply ui/package.json:23]: although it's still shouty

File: ui/cypress.json (hunk 2 above)
Comment (joefitter) [reply ui/cypress.json:2]: yessir `CYPRESS_baseUrl=[your baseUrl] npm run cypress:open` or `npm run cypress:open --config "baseUrl=myUrl"`

Comment (lennym) [reply ui/cypress.json:2]: I'll just leave this here then:
https://github.com/nhsx/standards-registry/blob/master/.github/workflows/deploy-dev.yml#L165
https://github.com/nhsx/standards-registry/blob/master/.github/workflows/deploy-test.yml#L114

File: ui/.babelrc [hunk 5]
```diff
@@ -1,4 +0,0 @@
-{
```
Comment (roc) [inline ui/.babelrc:1]: why is this deleted? Did we not need it?

Comment (joefitter) [reply ui/.babelrc:1]: next.js uses swc by default (instead of babel) - it's faster. Having a .babelrc turns it off

File: ui/package.json [hunk 6]
```diff
@@ -7,18 +7,20 @@
     "build": "next build",
     "start": "next start",
     "lint": "next lint",
-    "test:integration": "PORT=3000 wdio run ./wdio.conf.js",
+    "test:integration": "cypress run",
+    "cypress:open": "cypress open",
```
Comment (roc) [inline ui/package.json:11]: ooo what does this do? Interactive debugging?

Comment (joefitter) [reply ui/package.json:11]: yes that! try it out :)

File: ui/cypress/integration/standards/list.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    });
+
+    describe('Organisation mapping', () => {
+      it('Can search by organisation', () => {
+        cy.visit('/standards');
+        cy.get('input[name="q"]').type('prsb');
+
+        cy.contains('Search').click();
+
+        cy.get('#browse-results li').should('have.length', 6)
```
Comment (roc) [inline ui/cypress/integration/standards/list.js:39]: This test looks fine but the assertion doesn't :-/ There should be 16 results https://manage.test.standards.nhs.uk/organization/professional-record-standards-body

File: ui/cypress/integration/standards/list.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+      cy.get('#browse-results li').should('have.length', 1);
+    });
+
+    it('Can search by fuzzy match', () => {
+      cy.visit('/standards');
+      cy.get('input[name="q"]').type('alergy');
+
+      cy.contains('Search').click();
+
+      cy.get('#browse-results li').should('have.length', 1)
```
Comment (roc) [inline ui/cypress/integration/standards/list.js:29]: could you assert on the title including "Allergy"?
